### PR TITLE
NixOS module for server

### PR DESCRIPTION
**Pull request by aciceri**
I've close #1070 and re-opened it here using @brainrake branch since he was working on a different repo and then CI wasn't working.

### Pre-review checklist

- [ ] All code has been formatted using our config (`make format`)
- [ ] Any new API features or modification of existing behavior is covered as defined in our [test plan](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/test-plan.md)
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`), and the links to the appropriate issues/PRs have been included

### CODE REVIEW

**Review by klntsky — APPROVED** [review]

File: nix/ctl-server-nixos-module.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
```
Comment (klntsky) [inline nix/ctl-server-nixos-module.nix:1]: Can we have some CI check to ensure this code will be up to date?

Comment (aciceri) [reply nix/ctl-server-nixos-module.nix:1]: What do you mean?

Comment (klntsky) [reply nix/ctl-server-nixos-module.nix:1]: I mean, right now it's not actually executed in CI, right?

Comment (aciceri) [reply nix/ctl-server-nixos-module.nix:1]: It is, this PR also adds a new `nixosConfiguration` called `test` which is a minimal VM that makes use of the new NixOS modules. Moreover this `nixosConfiguration` is also added to `hydraJobs`.

**Pull request merged**